webapp/plugins/index.js

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+// EXPERIMENTAL - SUBJECT TO CHANGE
+
+import store from 'stores/redux_store.jsx';
+
+global.window.plugins = {};
+global.window.plugins.components = {};
+
+export function registerComponents(components) {
+    global.window.plugins.components = {...components, ...global.window.plugins.components};
+}
+
+export function initializePlugins() {
+    const pluginJson = global.window.mm_config.Plugins || '[]';
+    const pluginManifests = JSON.parse(pluginJson);
```

Comment:
We'll want a try/catch here. Also, do we want any sort of validation of the plugin manifest here?